sourcegit/config.py

```diff
@@ -28,6 +40,146 @@ def get_default_map_from_file() -> Optional[dict]:
+    @classmethod
+    def get_from_dict(cls, raw_dict: dict, validate=True) -> JobConfig:
+        if validate and not JobConfig.is_dict_valid(raw_dict):
+            raise Exception(f"Job config not valid.")
+
+        trigger_raw, release_to, metadata = exclude_from_dict(
+            raw_dict, "trigger", "release_to"
+        )
+        return JobConfig(
+            trigger=TriggerType[trigger_raw], release_to=release_to, metadata=metadata
+        )
+
+    @classmethod
+    def is_dict_valid(cls, raw_dict: dict) -> bool:
+        return Draft4Validator(JOB_CONFIG_SCHEMA).is_valid(raw_dict)
+
+
+@dataclass(unsafe_hash=True, frozen=True)
+class PackageConfig:
+    specfile_path: str
+    synced_files: List[str]
+    jobs: List[JobConfig]
+    metadata: dict
+
+    @classmethod
+    def get_from_dict(cls, raw_dict: dict, validate=True) -> PackageConfig:
+        if validate and not PackageConfig.is_dict_valid(raw_dict):
+            raise Exception("Package config not valid.")
+
+        specfile_path, synced_files, raw_jobs, metadata = exclude_from_dict(
+            raw_dict, "specfile_path", "synced_files", "jobs"
+        )
+
+        return PackageConfig(
+            specfile_path=specfile_path,
+            synced_files=synced_files,
+            jobs=[
+                JobConfig.get_from_dict(raw_job, validate=False) for raw_job in raw_jobs
+            ],
+            metadata=metadata,
+        )
+
+    @classmethod
+    def is_dict_valid(cls, raw_dict: dict) -> bool:
+        return Draft4Validator(PACKAGE_CONFIG_SCHEMA).is_valid(raw_dict)
+
+
+def get_local_package_config() -> Optional[PackageConfig]:
+    """
+    :return: local PackageConfig if present
+    """
+    for config_file_name in CONFIG_FILE_NAMES:
+
+        if os.path.isfile(config_file_name):
+            logger.debug(f"Local package config found: {config_file_name}")
+            try:
+                loaded_config = anymarkup.parse_file(config_file_name)
+            except Exception as ex:
+                logger.error(f"Cannot load package config '{config_file_name}'.")
+                raise Exception("Cannot load package config.", ex)
+
+            try:
+                package_config = PackageConfig.get_from_dict(
+                    raw_dict=loaded_config, validate=True
+                )
+                return package_config
+            except Exception as ex:
+                logger.error(
+                    f"Cannot parse package config:\n{json.dumps(loaded_config, indent=4)}"
+                )
+                raise Exception("Package config is not valid.", ex)
+
+    return None
+
+
+def get_packit_config_from_repo(
+        sourcegit_project: GitProject, branch: str
+) -> PackageConfig:
+    for config_file_name in CONFIG_FILE_NAMES:
+        try:
+            config_file = sourcegit_project.get_file_content(
+                path=config_file_name, ref=branch
+            )
+        except FileNotFoundError:
+            continue
+
+        try:
```

Comment:
actually these two inner blocks are identical with the function above, can we refactor?

Reply:
Not exactly (`anymarkup.parse` x `anymarkup.parse_file`), but I can tweak it a little.

Reply:
resolved